### PR TITLE
Go back to 22 to be crash free

### PR DESCRIPTION
**Pull request by nirvn**
Gives me crash-free local APKs again, and ~~hopefully~~ the unwind flags will work on armv7.

Edit: @marioba confirmed the APKs here are crash-free again.

### CODE REVIEW

File: cmake/qgis-cmake-wrapper.cmake [hunk 1]
```diff
@@ -107,7 +107,10 @@ if(TRUE) # Should possibly have a "static only" check
   if(ANDROID)
     find_library(libdl dl)
     get_filename_component(arch_path ${libdl} DIRECTORY)
-    set(pkgcfg_lib_PC_SPATIALITE_c++ "${arch_path}/libc++.so")
+    set(pkgcfg_lib_PC_SPATIALITE_c++ "${arch_path}/${ANDROID_PLATFORM_LEVEL}/libc++.so")
+    if(NOT EXISTS ${pkgcfg_lib_PC_SPATIALITE_c++})
```
Comment (nirvn) [inline cmake/qgis-cmake-wrapper.cmake:111]: @m-kuhn , see failure here (https://github.com/opengisch/QField/actions/runs/3065501184/jobs/4949665264) as to the why this hack was introduced.

File: vcpkg/overlay/json-c/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,24 @@
+vcpkg_from_github(
```
Comment (nirvn) [inline vcpkg/overlay/json-c/portfile.cmake:1]: This is needed due to the following failure: https://github.com/opengisch/QField/actions/runs/3065670252/jobs/4950004479

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 30dd344efde76e1a6cb2f07637094f1309d30560: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3346-44b0d46cf4b4fe061c7844a2b97fbda2d11e1b63-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3346-44b0d46cf4b4fe061c7844a2b97fbda2d11e1b63-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3346-44b0d46cf4b4fe061c7844a2b97fbda2d11e1b63-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3346-44b0d46cf4b4fe061c7844a2b97fbda2d11e1b63-x86-android.apk)
</details>

File: vcpkg/triplets/arm-android.cmake [hunk 3]
```diff
@@ -4,5 +4,7 @@ set(VCPKG_LIBRARY_LINKAGE static)
 set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_BUILD_TYPE release)
 
-set(VCPKG_CXX_FLAGS "-fstack-protector-strong")
-set(VCPKG_C_FLAGS "-fstack-protector-strong")
+set(ENV{VCPKG_ANDROID_NATIVE_API_LEVEL} "detect")
```
Comment (m-kuhn) [inline vcpkg/triplets/arm-android.cmake:7]:
```suggestion
```

File: vcpkg/triplets/arm64-android.cmake [hunk 4]
```diff
@@ -4,5 +4,7 @@ set(VCPKG_LIBRARY_LINKAGE static)
 set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_BUILD_TYPE release)
 
+set(ENV{VCPKG_ANDROID_NATIVE_API_LEVEL} "detect")
```
Comment (m-kuhn) [inline vcpkg/triplets/arm64-android.cmake:7]:
```suggestion
```

File: cmake/qgis-cmake-wrapper.cmake (hunk 1 above)
Comment (m-kuhn) [reply cmake/qgis-cmake-wrapper.cmake:111]: If the build succeeds it's all good 👍

**Review by m-kuhn — APPROVED** [review]
If it fixes something it fixes something ...

**Comment by m-kuhn** [comment]
Oops that last one was unintended and no keyboard here to undo 🙈

**Comment by m-kuhn** [comment]
worth backporting?

**Comment by nirvn** [comment]
@m-kuhn , no, we're still on r22 on the release-2_3 branch. This was (fortunately) a master-only regression.

**Pull request merged**